integrationTesting/tests/organize/journeys/instance-detail/sidebar.spec.ts

```diff
@@ -457,15 +457,13 @@ test.describe('Journey instance detail page sidebar', () => {
     ]);
 
     await page.locator(`text="${ActivistTag.title}"`).hover();
-
-    await Promise.all([
-      page.waitForResponse(
-        `**/orgs/${KPD.id}/journey_instances/${ClarasOnboarding.id}/tags/${ActivistTag.id}`
-      ),
-      page.locator(`[data-testid=TagChip-deleteButton]`).first().click(),
-    ]);
+    const deleteButtonTransition = page.waitForTimeout(500);
+    await deleteButtonTransition;
+    await page.locator(`[data-testid=TagChip-deleteButton]`).first().click();
 
     // Expect to have made request to delete tag
-    expect(deleteTagLog().length).toEqual(1);
+    await expect(async () => {
+      expect(deleteTagLog()).toHaveLength(1);
+    }).toPass();
```

Comment:
I don't understand how the inner function here is async. I mean, I see the `async` keyword, but the function does not return anything, nor does it `await` anything, so unless I'm mistaken it will just return immediately. It's basically just a function that wraps `expect()` in a promise. Why is that necessary?